From libc-ports-return-1240-listarch-libc-ports=sources dot redhat dot com at sourceware dot org Sat May 16 14:40:07 2009
Return-Path: <libc-ports-return-1240-listarch-libc-ports=sources dot redhat dot com at sourceware dot org>
Delivered-To: listarch-libc-ports at sources dot redhat dot com
Received: (qmail 13552 invoked by alias); 16 May 2009 14:40:06 -0000
Received: (qmail 13539 invoked by uid 22791); 16 May 2009 14:40:05 -0000
X-SWARE-Spam-Status: No, hits=-2.3 required=5.0 	tests=AWL,BAYES_00,SPF_PASS
X-Spam-Check-By: sourceware.org
Received: from mail.codesourcery.com (HELO mail.codesourcery.com) (65.74.133.4)     by sourceware.org (qpsmtpd/0.43rc1) with ESMTP; Sat, 16 May 2009 14:40:00 +0000
Received: (qmail 18469 invoked from network); 16 May 2009 14:39:58 -0000
Received: from unknown (HELO digraph.polyomino.org.uk) (joseph@127.0.0.2)   by mail.codesourcery.com with ESMTPA; 16 May 2009 14:39:58 -0000
Received: from jsm28 (helo=localhost) 	by digraph.polyomino.org.uk with local-esmtp (Exim 4.69) 	(envelope-from <joseph@codesourcery.com>) 	id 1M5L3N-0002qI-0a 	for libc-ports@sourceware.org; Sat, 16 May 2009 14:39:57 +0000
Date: Sat, 16 May 2009 14:39:57 +0000 (UTC)
From: "Joseph S dot  Myers" <joseph at codesourcery dot com>
To: libc-ports at sourceware dot org
Subject: [ARM,MIPS,M68K] fallocate64 exports
Message-ID: <Pine.LNX.4.64.0905161432560.9931@digraph.polyomino.org.uk>
MIME-Version: 1.0
Content-Type: TEXT/PLAIN; charset=US-ASCII
Mailing-List: contact libc-ports-help at sourceware dot org; run by ezmlm
Precedence: bulk
List-Id: <libc-ports.sourceware.org>
List-Subscribe: <mailto:libc-ports-subscribe at sourceware dot org>
List-Post: <mailto:libc-ports at sourceware dot org>
List-Help: <mailto:libc-ports-help at sourceware dot org>, <http://sourceware dot org/lists dot html#faqs>
Sender: libc-ports-owner at sourceware dot org
Delivered-To: mailing list libc-ports at sourceware dot org

As per <http://sourceware.org/ml/libc-hacker/2009-05/msg00003.html> ports 
architectures not using sysdeps/unix/sysv/linux/wordsize-64 need to export 
fallocate64 at version GLIBC_2.11 in their own Versions files.  I've 
applied these ARM and MIPS fixes; OK to push the M68K fix?

diff --git a/sysdeps/unix/sysv/linux/arm/Versions b/sysdeps/unix/sysv/linux/arm/Versions
index 2ddb2af..1d9e964 100644
--- a/ports/sysdeps/unix/sysv/linux/arm/Versions
+++ b/ports/sysdeps/unix/sysv/linux/arm/Versions
@@ -34,4 +34,7 @@ libc {
   GLIBC_2.3.3 {
     posix_fadvise64; posix_fallocate64;
   }
+  GLIBC_2.11 {
+    fallocate64;
+  }
 }
diff --git a/sysdeps/unix/sysv/linux/mips/Versions b/sysdeps/unix/sysv/linux/mips/Versions
index 09df42d..a56322a 100644
--- a/ports/sysdeps/unix/sysv/linux/mips/Versions
+++ b/ports/sysdeps/unix/sysv/linux/mips/Versions
@@ -34,4 +34,7 @@ libc {
     # _*
     _test_and_set;
   }
+  GLIBC_2.11 {
+    fallocate64;
+  }
 }
diff --git a/ChangeLog.m68k b/ChangeLog.m68k
index c68e94a..3afc3da 100644
--- a/ports/ChangeLog.m68k
+++ b/ports/ChangeLog.m68k
@@ -1,3 +1,8 @@
+2009-05-16  Joseph Myers  <joseph@codesourcery.com>
+
+	* sysdeps/unix/sysv/linux/m68k/Versions (libc): Add
+	fallocate64@@GLIBC_2.11.
+
 2009-04-25  Andreas Schwab  <schwab@linux-m68k.org>
 
 	* sysdeps/unix/sysv/linux/m68k/kernel-features.h: Revert last
diff --git a/sysdeps/unix/sysv/linux/m68k/Versions b/sysdeps/unix/sysv/linux/m68k/Versions
index 0799bf3..5650f7f 100644
--- a/ports/sysdeps/unix/sysv/linux/m68k/Versions
+++ b/ports/sysdeps/unix/sysv/linux/m68k/Versions
@@ -29,4 +29,7 @@ libc {
     # v*
     versionsort64;
   }
+  GLIBC_2.11 {
+    fallocate64;
+  }
 }

-- 
Joseph S. Myers
joseph@codesourcery.com

